org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/apps/graph/WPAWidget.java

```diff
@@ -0,0 +1,67 @@
+package org.adempiere.webui.apps.graph;
+
+import org.adempiere.webui.apps.graph.WPerformanceIndicator.Options;
+import org.adempiere.webui.component.Panel;
+import org.compiere.model.MGoal;
+import org.compiere.util.CLogger;
+import org.zkoss.zk.ui.event.Event;
+import org.zkoss.zk.ui.event.EventListener;
+import org.zkoss.zk.ui.event.Events;
+import org.zkoss.zul.Div;
+import org.zkoss.zul.Label;
+
+public class WPAWidget extends Panel implements EventListener<Event> {
+
+	/**
+	 * 
+	 */
+	private static final long serialVersionUID = 5684412399657327777L;
+	
+	/**************************************************************************
+	 * 	Constructor
+	 */
+	public WPAWidget (MGoal goal, Options options)
+	{
+		super();
+		goal.updateGoal(false);
+		init(goal, options);
+	}
+	
+	/**	Logger	*/
+	private static final CLogger log = CLogger.getCLogger (WPAWidget.class);
+	
+	/**
+	 * 	Static/Dynamic Init
+	 * @param goal
+	 * @param options 
+	 */
+	private void init(MGoal goal, Options options)
+	{
+		Div div = new Div();
+		appendChild(div);
+		WPerformanceIndicator pi = new WPerformanceIndicator(goal, options);
+		div.appendChild(pi);
+		pi.addEventListener(Events.ON_CLICK, this);
```

Comment:
not a must but this can actually be simplify to `if (goal.getMeasure() != null) pi.addEventListener(Events.ON_CLICK, e -> new WPerformanceDetail(pi.getGoal());`